model/runner/ProctoringRunnerService.php

```diff
@@ -39,30 +39,31 @@ class ProctoringRunnerService extends QtiRunnerService
     use OntologyAwareTrait;
 
     /**
-     * Get Test Context.
+     * Get Test Data.
      *
      * @param RunnerServiceContext $context
      * @return array
      * @throws \common_Exception
      * @throws \common_exception_NotFound
      * @throws \core_kernel_persistence_Exception
      */
-    public function getTestContext(RunnerServiceContext $context)
+    public function getTestData(RunnerServiceContext $context)
     {
-        $response = parent::getTestContext($context);
+        $testContext = parent::getTestContext($context);
```

Comment:
Instead of retrieving the whole testContext, please call directly `$this->getServiceManager()->get(SectionPauseService::SERVICE_ID)->couldBePaused($context->getTestSession())` (or something like that) in order to get the value of `sectionPause`. 
And check if the test state is already in the interacting state  to support the proctoring implementation https://github.com/oat-sa/extension-tao-proctoring/blob/7cd8c4224a54ca00fb00ccef8bc820b14b7bd94c/model/execution/ProctoredSectionPauseService.php#L52

Reply:
done in latest commit